stac_fastapi/api/tests/test_app.py

```diff
@@ -0,0 +1,149 @@
+import importlib
+import os
+from datetime import datetime
+from typing import List, Optional, Union
+
+import pytest
+from fastapi.testclient import TestClient
+from pydantic import BaseModel
+
+from stac_fastapi.api.app import StacApi
+from stac_fastapi.api.models import create_get_request_model, create_post_request_model
+from stac_fastapi.extensions.core.filter.filter import FilterExtension
+from stac_fastapi.types import core, response_model, search
+from stac_fastapi.types.config import ApiSettings
+from stac_fastapi.types.search import BaseSearchPostRequest
+
+
+@pytest.fixture
+def cleanup():
+    old_environ = dict(os.environ)
+    yield
+    os.environ.clear()
+    os.environ.update(old_environ)
+
+
+@pytest.mark.parametrize(
+    "validate, response_type",
+    [
+        ("True", BaseModel),
+        ("False", dict),
+    ],
+)
+def test_app(validate, response_type, collection_dict, item_dict, cleanup):
+    os.environ["VALIDATE_RESPONSE"] = str(validate)
```

Comment:
could we use monkeypatch here instead of writing the env variable 
https://docs.pytest.org/en/6.2.x/monkeypatch.html